addons/account_bank_statement_extensions/account_bank_statement.py

```diff
@@ -1,135 +1,91 @@
-#
-#    Copyright (c) 2011 Noviat nv/sa (www.noviat.be). All rights reserved.
-#
-#    This program is free software: you can redistribute it and/or modify
-#    it under the terms of the GNU Affero General Public License as
-#    published by the Free Software Foundation, either version 3 of the
-#    License, or (at your option) any later version.
-#
-#    This program is distributed in the hope that it will be useful,
-#    but WITHOUT ANY WARRANTY; without even the implied warranty of
-#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-#    GNU Affero General Public License for more details.
-#
-#    You should have received a copy of the GNU Affero General Public License
-#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
-#
-##############################################################################
 
-import time
-from openerp.osv import fields, osv
-import openerp.addons.decimal_precision as dp
-from openerp.tools.translate import _
+from openerp import models, fields, api, _
 from openerp.exceptions import UserError
 
-class account_bank_statement(osv.osv):
+
+class AccountBankStatement(models.Model):
     _inherit = 'account.bank.statement'
 
-    def write(self, cr, uid, ids, vals, context=None):
-        if context is None:
-            context = {}
+    @api.multi
+    def write(self, vals):
         # bypass obsolete statement line resequencing
-        if vals.get('line_ids', False) or context.get('ebanking_import', False):
-            res = super(osv.osv, self).write(cr, uid, ids, vals, context=context)
+        if vals.get('line_ids') or self._context.get('ebanking_import'):
+            res = super(models.Model, self).write(vals)
         else:
-            res = super(account_bank_statement, self).write(cr, uid, ids, vals, context=context)
+            res = super(AccountBankStatement, self).write(vals)
         return res
 
-    def button_confirm_bank(self, cr, uid, ids, context=None):
-        bank_statement_line_obj = self.pool.get('account.bank.statement.line')
-        super(account_bank_statement, self).button_confirm_bank(cr, uid, ids, context=context)
-        for st in self.browse(cr, uid, ids, context=context):
+    @api.multi
+    def button_confirm_bank(self):
+        BankStatementLine = self.env['account.bank.statement.line']
+        super(AccountBankStatement, self).button_confirm_bank()
+        for st in self:
             if st.line_ids:
                 line_ids = [l.id for l in st.line_ids]
-                cr.execute("UPDATE account_bank_statement_line  \
-                    SET state='confirm' WHERE id in %s ",
-                    (tuple(line_ids),))
-                bank_statement_line_obj.invalidate_cache(cr, uid, ['state'], line_ids, context=context)
-        return True
+                self.env.cr.execute("UPDATE account_bank_statement_line  \
+                    SET state='confirm' WHERE id in %s ", (tuple(line_ids),))
```

Comment:
SQL query should be rewritten with the new API, unless there is a comment stating this query is needed !!!